build/parser.rs

```diff
@@ -1,61 +1,542 @@
-use std::default::Default;
-use std::cmp::Ordering;
-use std::io::{Write, Read};
 use crc16;
+use std::cmp::Ordering;
+use std::default::Default;
+use std::io::{Read, Write};
 
 use xml::reader::{EventReader, XmlEvent};
 
-#[derive(Debug, PartialEq, Clone)]
+use quote::{Ident, Tokens};
+
+#[derive(Debug, PartialEq, Clone, Default)]
+pub struct MavProfile {
+    pub includes: Vec<String>,
+    pub messages: Vec<MavMessage>,
+    pub enums: Vec<MavEnum>,
+}
+
+impl MavProfile {
+    /// Go over all fields in the messages, and if you encounter an enum,
+    /// update this enum with information about whether it is a bitmask, and what
+    /// is the desired width of such.
+    fn update_enums(mut self) -> Self {
+        for msg in &self.messages {
+            for field in &msg.fields {
+                if let Some(ref enum_name) = field.enumtype {
+                    // it is an enum
+                    if let Some(ref dsp) = field.display {
+                        // it is a bitmask
+                        if dsp == "bitmask" {
+                            // find the corresponding enum
+                            for mut enm in &mut self.enums {
+                                if enm.name == *enum_name {
+                                    // this is the right enum
+                                    enm.bitfield = Some(field.mavtype.rust_type());
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        self
+    }
+
+    /// If we are not using Mavlink v2, remove messages with id's > 254
+    #[cfg(not(feature="mavlink2"))]
+    fn update_messages(mut self) -> Self {
+        println!("Updating messages");
+        let msgs = self.messages.into_iter().filter(|x| x.id <= 254).collect::<Vec<MavMessage>>();
+        self.messages = msgs;
+        self
+    }
+
+    /// Simple header comment
+    fn emit_comments(&self) -> Ident {
+        Ident::from(format!(
+            "// This file was automatically generated, do not edit \n"
```

Comment:
You don't even need to misuse `Ident` for this, just write it to the output file before the source code.